### DISCUSSION TITLE
How to copy and use an home/user as the default template for Raspberry Pi OS clients ?

### DISCUSSION THREAD

**Discussion by ThierryM** [ltsp/ltsp#345] [Q&A] [answered]
Hi,
I have an home/model under Raspberry Pi OS. How can I copy and use this /home/model in chroot in order that the clients have the same environment ?
I've read the wiki with the 2 pages for the guest clients but I think that they not respond to my problem.
Regards,

Thierry

**Comment by alkisg** [comment]
Do you mean that you have created a /home/template user, and you want all new users to be based on the same template?

If so, type these commands:

```
cd /home/template
grep -r template .
```

You should see the "template" username in various configuration files, binary dconf settings etc.
Now create a script that will copy that directory for all your user accounts, while replacing the "template" word with the new usernames:

```
cp -a /home/template /home/newuser
TODO: replace template newuser in /home/newuser
```

Note the "TODO command there. It's no easy task, because you cannot go ahead and replace it in binary files like dconf.

What I meant with all the above, is that "use a template home directory" CANNOT be done properly. It needs to be a manual process, where the sysadmin carefully examines where the "template" username and path appears inside the template home.

With guest accounts, it's different, because they are all named "guest" and they all have the same home directory, /home/guest, so there's no need to replace anything.

**Comment by ThierryM** [comment]
Thanks @alkisg for your help.
Yes, on my Raspberry Pi OS image I have the user "template" with his home directory "/home/template". I modify in NFS-RW mode.
My problem (as I undestand) is that the /home/template is in the chroot image of Raspberry Pi OS. How can I transfer this /home/template on the Ltsp server for the others users (they can be guests too for simpler deployment)  ?
Regards

**Comment by alkisg** [comment]
The commands to transfer a home directory from NFSRW mode to the normal LTSP mode, are:

```shell
adduser newuser
rsync -a /srv/ltsp/raspios/home/template/ /home/newuser
chown -R newuser:newuser /home/newuser
```

But please read again what I wrote in my previous message, why you should NOT do that.

**Comment by ThierryM** [comment]
So I could transfer my /srv/ltsp/raspios/home/template/ in /home/guest and follow the wiki instructions for guest ?

**Comment by alkisg** [comment]
You could transfer /srv/ltsp/raspios/home/template/ to /home/template. You cannot rename home directories without problems, that is what I meant.

**Comment by ThierryM** [comment]
Ok, I think I understood ;-). Thanks.
In fact my problem is to propose for the clients users the same desktop as my user template. It's for that reason that I thought to copy the /home/template directory .
So the problem should be rather how to copy the /home/template desktop on the chroot image to etc/skel that is used for new users in order they have the same GUI.

**Comment by alkisg** [comment]
The users in LTSP mode are NOT in the chroot.
The home directories in LTSP mode are NOT in the chroot.
They both are in the server.
You add users to the server.
You add home directories to the server.
See the commands I wrote there: https://github.com/ltsp/ltsp/discussions/345#discussioncomment-148249

**Comment by ThierryM** [comment]
I know that all the users and their home directory are on the server.
But as I work with an armhf image (Raspberry Pi OS) for the clients, I have to make my settings on the chroot image in NFS_RW mode. I can't make directly on the server.
When I add a new user, it's on the server and the defauts settings seems to come from /etc/skel of the server, no ?
I thought that with this method, no more problem to rename the home/directory because the new users would have their own home with the default settings contained in /etc/skel of the server. Where is my mistake ?

**Comment by alkisg** [comment]
The restriction that "you can't rename home directories without risking broken settings" comes from all the programs that you use; your desktop environment, your browsers, your dconf settings etc. It is not related to LTSP. It affects all Linux installations and all architectures.

Normally /etc/skel is almost empty. If you put things inside it, you need to make sure that they don't contain the "template" username anywhere inside it. You can't just copy /home/template into /etc/skel and expect it to work.
The architecture isn't related, /etc/skel usually doesn't mention the architecture at all.
If you prepare a correct /etc/skel, then you can just use `adduser` on the server, you don't have to boot clients and prepare things there.

**Comment by ThierryM** [comment]
Yes, the solution is to create a correct /etc/skel (with .config) and relative path (and not harcoded path with /home/template/) with the variable $HOME). I'll give a try.
Thanks for your help.

**Comment by ThierryM** [comment, marked as answer]
It works !
Here is the method I have used to clone the /home/administrateur in the Raspberry Pi OS based distribution Primtux4-Rpi4 :
- First I have prepared my Primtux4-Rpi4 on a classic Raspberry Pi 4 alone.
- Then I have installed the image on the Ltsp server (according the instructions on https://ltsp.org for Raspberry PI OS installation)
- I've personnalized the user "administrateur" in NFS_RW mode from a raspberry pi 4 client and ended the image construction with the LTSP mode netbooting.
- then I used this command `rsync -a /srv/ltsp/primtux4-rpi4/home/administrateur/ /etc/skel` to duplicate the parameters.
- I've cleaned the /etc/skel searching the path hardcoded /home/administrateur and replacing by the $HOME variable.
- i've created a new user in the server.
- I ended with the command `ltsp initrd` on the server to add this new user for ltsp.
Now I can boot my Raspberry Pi OS and when I'm logging as the user created, i have the same environment desktop as my originally administrateur in Primtux.
All seems working well !
